mlrun/api/utils/auth/providers/opa.py

```diff
@@ -241,8 +250,14 @@ def _generate_filter_request_body(
         body = {
             "input": {
                 "resources": resources,
-                "action": action,
+                "action": str(action),
                 "ids": auth_info.get_member_ids(),
             },
         }
         return body
+
+    async def _ensure_session(self):
```

Comment:
Does this method really need to be async? IIUC it doesn't do anything interesting other than initialize the in-memory objects, so why add the context-switch overhead on this?

Reply:
it must be async. see https://stackoverflow.com/questions/52232177/runtimeerror-timeout-context-manager-should-be-used-inside-a-task